src/app/resbumit/resubmitRequestHandler.ts

```diff
@@ -68,29 +69,34 @@ export class ResubmitRequestHandler {
 
 
   async getRecentResubmissions() {
-    const recents = await this.dynamoDBClient.send(new ScanCommand({
+    const recents = await this.dynamoDBClient.send(new QueryCommand({
       TableName: process.env.RESUBMISSION_TABLE_NAME,
+      ScanIndexForward: false, // true = ascending, false = descending
       Limit: 30,
+      KeyConditionExpression: 'id = :id',
+      ExpressionAttributeValues: {
+        ':id': { S: RESUBMISSION },
+      },
     }));
     const mapped = recents.Items?.map(item => {
       return {
         ref: item.reference.S,
         timestamp: item.timestamp.S,
+        user: item.user?.S ?? '-',
       };
     });
-    return mapped?.sort((a: any, b: any) => {
-      return b.timestamp?.localeCompare(a.timestamp);
-    });
+    return mapped;
   }
 
-  async addToRecentResubmissions(reference: string) {
+  async addToRecentResubmissions(reference: string, user: string) {
     const timestamp = new Date().toISOString();
     const ttl = Date.now() + (1000 * 3600 * 24 * 30); // 30 days
     await this.dynamoDBClient.send(new PutItemCommand({
       Item: {
-        id: { S: randomUUID() },
+        id: { S: RESUBMISSION },
```

Comment:
Gaat dit goed? Bevat de composite key ook het timestamp? Anders is de key niet uniek

Reply:
Waarom doen we dit?

Reply:
Ik dénk dat het wel werkt, met als nadeel dat alles in dezelfde dynamodb partition terecht zou komen. Gezien de hoeveelheid data mogelijk geen probleem.